### PR TITLE
Default NO_REPLY_ADDRESS to DOMAIN

### LINKED ISSUE #9275: NoReply Address Always Dummy Address Regardless of Root URL Domain
- Gitea version (or commit ref): 1.9.0
- Git version: 2.17.1
- Operating system: Ubuntu 18.04
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:
(not relevant)

## Description
Installing Gitea set `NO_REPLY_ADDRESS` option on `app.ini` to dummy `noreply@example.org` address, regardless of domain in `ROOT_URL`. For NoReply emails to function correctly, use address from same domain as in `ROOT_URL`. For example, if `ROOT_URL` domain is `mygitea.instance`, `NO_REPLY_ADDRESS` should be `noreply@mygitea.instance`.

## Screenshots
(not relevant)

### PR DESCRIPTION
**Pull request by zeripath**
The default NO_REPLY_ADDRESS was previously noreply.example.org
This PR changes this to default to the default DOMAIN for gitea.

Fixes #9275

### CODE REVIEW

**Comment by zeripath** [comment]
This may not be the completely correct thing to do as it potentially just reveals the user's email.

File: docs/content/doc/advanced/config-cheat-sheet.en-us.md [hunk 1]
```diff
@@ -300,7 +300,7 @@ relation to port exhaustion.
 - `CAPTCHA_TYPE`: **image**: \[image, recaptcha\]
 - `RECAPTCHA_SECRET`: **""**: Go to https://www.google.com/recaptcha/admin to get a secret for recaptcha.
 - `RECAPTCHA_SITEKEY`: **""**: Go to https://www.google.com/recaptcha/admin to get a sitekey for recaptcha.
-- `RECAPTCHA_URL`: **https://www.google.com/recaptcha/**: Set the recaptcha url - allows the use of recaptcha net.
+- `RECAPTCHA_URL`: **https://www.google.com/rec**aptcha/**: Set the recaptcha url - allows the use of recaptcha net.
```
Comment (lunny) [inline docs/content/doc/advanced/config-cheat-sheet.en-us.md]: Why change this.

Comment (zeripath) [reply docs/content/doc/advanced/config-cheat-sheet.en-us.md]:
```suggestion
- `RECAPTCHA_URL`: **https://www.google.com/recaptcha/**: Set the recaptcha url - allows the use of recaptcha net.
```

Comment (zeripath) [reply docs/content/doc/advanced/config-cheat-sheet.en-us.md]: Oops that was a mistake

**Comment by mrsdizzie** [comment]
If the persons email address is also their username@domain then it would defeat the purpose.

It also sounds like the linked issue expects this to be resolved by having the private address be ```noreply@domain``` but Gitea still uses the username in  a 'private address' so I think just setting NO_REPLY to DOMAIN causes a new problem and probably doesn't actually fix what the person in that issue wants or expects (to have the address end up as ```noreply@domain``` when private).

I think part of the confusion is this uses a bad variable name -- it sounds like the issue reporter expects ```NO_REPLY_ADDRESS```to be an actual full/single e-mail address that will be used in all cases. Of course, it is just a domain name that is used in building the address. So their issue is that they think ```NO_REPLY_ADDRESS```should be the *address* ```noreply@domain``` when it really isn't an address at all.

**Comment by zeripath** [comment]
@mrsdizzie that's what I meant by it being not necessarily the correct thing.

I would say that this does at least hide the private email address if they're not in the current domain - however the current system private email system isn't very good at hiding emails in the case that their email address is username@domain in any case as you can easily back generate that from the NO_REPLY_ADDRESS email as username@NO_REPLY_ADDRESS -> username@domain.

Yeah I agree the configuration name is poorly named - I think even gitea.com had it wrong at some point.

The email address should probably be user-:userid:@NO_REPLY_ADDRESS

**Comment by mrsdizzie** [comment]
Agree this is all confusing and also conflating two separate ideas: a 'private' address and an address that indicates via its name that you shouldn't/can't reply to it because it isn't valid.

I think the address it self should be invalid *and* indicate no reply by default since it is likely the domain name will have also valid MX records. The default now doesn't have valid MX records and can't have mail sent to it which I think is correct.

Say gitea.com/domain has a catch-all enabled and now it makes every 'private' address a valid one that will accept and deliver mail to somewhere.

Or say you have an existing email address zeripath@gitea.com and I sign up for the username zeripath on gitea.com and mark as private and now all my commits say zeripath@gitea.com. 

It seems that just using a combination of username + domain name will cause more potential problems then it fixes. I'm also not sure it qualifies as a noreply address if it can actually work and have mail sent to it in any case. Maybe that is just more trouble of trying to combine noreply and private.

Perhaps defaulting to noreply.domain would solve some of that or provide a better default.

**Comment by mrsdizzie** [comment]
In think this is also what the linked issue is requesting as well, just reversed. They expect it to be noreply@domain (which is presumably invalid).

**Review by techknowlogick — APPROVED** [review]

File: custom/conf/app.ini.sample [hunk 2]
```diff
@@ -504,7 +504,7 @@ DEFAULT_ALLOW_ONLY_CONTRIBUTORS_TO_TRACK_TIME = true
 ; Default value for the domain part of the user's email address in the git log
 ; if he has set KeepEmailPrivate to true. The user's email will be replaced with a
 ; concatenation of the user name in lower case, "@" and NO_REPLY_ADDRESS.
-NO_REPLY_ADDRESS = noreply.example.org
+NO_REPLY_ADDRESS = %(DOMAIN)s
```
Comment (zeripath) [inline custom/conf/app.ini.sample]:
```suggestion
NO_REPLY_ADDRESS = noreply.%(DOMAIN)s
```

File: modules/setting/service.go [hunk 3]
```diff
@@ -83,7 +83,7 @@ func newService() {
 	Service.DefaultEnableDependencies = sec.Key("DEFAULT_ENABLE_DEPENDENCIES").MustBool(true)
 	Service.AllowCrossRepositoryDependencies = sec.Key("ALLOW_CROSS_REPOSITORY_DEPENDENCIES").MustBool(true)
 	Service.DefaultAllowOnlyContributorsToTrackTime = sec.Key("DEFAULT_ALLOW_ONLY_CONTRIBUTORS_TO_TRACK_TIME").MustBool(true)
-	Service.NoReplyAddress = sec.Key("NO_REPLY_ADDRESS").MustString("noreply.example.org")
+	Service.NoReplyAddress = sec.Key("NO_REPLY_ADDRESS").MustString(Domain)
```
Comment (zeripath) [inline modules/setting/service.go]:
```suggestion
	Service.NoReplyAddress = sec.Key("NO_REPLY_ADDRESS").MustString("noreply." + Domain)
```

**Comment by zeripath** [comment]
@mrsdizzie yeah I think that that is more correct. I've changed it to default to `noreply.DOMAIN`

**Review by mrsdizzie — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Pull request merged**